### DISCUSSION TITLE
[Feature Request/Bugfix?] Add notifications for tickets on achievements you've worked on not made by you

### DISCUSSION THREAD

**Discussion by UnliquidSnake** [RetroAchievements/RAWeb#2364] [Ideas]
So, when you get a ticket on your own cheevo, you get a big honking site notification. But no such thing for ones that you've revised but were originally made by other devs.

I'd like some way to notify devs that an achievement they changed the logic for now has a ticket. Right now there are none on the site that I know of. I am subscribed to tickets on every set I've worked on, full set, revision or otherwise, yet I don't get anything when a ticket on a revision shows up - no site message, no email, nothing. I just recently solved two tickets for achievements that aren't my own, and the only way to see they exist is look on the game page -> Open Tickets, which is going to be a lot harder when you have like 30-40 games to keep track of.

What I propose is:

1) Give a site notification for a ticket if the user is subscribed to tickets on that particular set which the achievement belongs to, regardless of their status or who made the achievements in the set.
2) Give a site notification for a ticket if the user was the last person who changed the logic of that particular achievement. This is tracked on the achievement pages and most likely saved somewhere in a database, I think this is reasonable so that you can't just make a mistake changing an achievement, break it and forget about it until some guy sends you an angry PM or comment. I've had this happen a few times, it sucks.
-> Optional: For 2), obligate the user to solve said tickets as though they were their own (ban for 2 month hanging period and all). I'm not sure how feasible this is considering ticket busters aren't going to remember every single one they've fixed, and it might dissuade them from fixing them in the first place, but it would be an added measure to enforce responsibility for the stuff you do decide to fix. This needs further discussion for the full implications of what that would be like, I'm not too sure myself.

**Comment by Jamiras** [comment]
You can subscribe to a game's tickets via the dev box on the game page:
![image](https://github.com/RetroAchievements/RAWeb/assets/32680403/5e24f8e4-b7c8-455b-9dfb-f0aa5a0828eb)

But this only sends emails.

The entire site notification system needs to be reworked. See #1892

**Reply by UnliquidSnake** [reply]
Please read my initial post. I am subscribed to several sets I've worked on as you showed and it's never sent me anything of the sort. Maybe it has an issue with certain hosts or something (Gmail), but that seems unlikely.

**Reply by Jamiras** [reply]
> I don't get anything when a ticket on a revision shows up - no site message, no email, nothing. 

Tickets don't generate site messages, and you have all email messages turned off.

As there isn't an individual setting for enabling ticket emails, and tickets used to generate private messages, it's tied to the "Private Message / Email" checkbox.

**Reply by UnliquidSnake** [reply]
They do when you're the original creator of that specific achievement, and only in that case. I think they should also do this in the cases I've outlined above.

**Reply by Jamiras** [reply]
They don't create site messages. That was removed by #2083 back in December.

Are you referring to the link under the bell that takes you to the list of tickets assigned to you? That has nothing to do with subscriptions.

**Reply by UnliquidSnake** [reply]
Now that the new site update added a separate navigation thing for tickets, any plans for making this work?

**Reply by drisc** [reply]
As Jamiras has said Tickets don’t generate Site Messages. There are plans in the works to update our notification system to use the Laravel Notifiable trait, this will include a complete overhaul and expansion of the types of interactions that can produce a notification for a user.

We’ll keep this in mind as a feature request though it may not be in the initial notifications rework release.
